src/Util/DotEnvLoader.php

```diff
@@ -0,0 +1,124 @@
+<?php
+namespace SilverStripe\Upgrader\Util;
+
+use M1\Env\Parser;
+use SilverStripe\Upgrader\CodeCollection\CodeChangeSet;
+
+/**
+ * Load an existing `.env` file and provide a way for merging it with a new set of value.
+ *
+ * Loads environment variables from .env files
+ *
+ */
+class DotEnvLoader
+{
+
+    /**
+     * Full path to an env file that is read (if available). That's also where the results will be written.
+     * @var string
+     */
+    private $envFilePath;
+
+    /**
+     * Existing content of the .env file.
+     * @var [type]
+     */
+    private $inputContent = '';
+
+    /**
+     * Content that will be outputted to the .env file.
+     * @var [type]
+     */
+    private $outputContent;
+
+    /**
+     * @param string $envFilePath File Path to `.env` file
+     * @param array  $consts List of variables that should be added to envrionment file.
+     */
+    public function __construct(string $envFilePath, array $consts)
+    {
+        $this->envFilePath = $envFilePath;
+
+        $this->inputContent = $this->readFile();
+        $outputConsts = $this->parseFile($consts);
+        $this->outputContent = $this->buildOutput($outputConsts);
```

Comment:
Should we merge both input and output env files? If we have a var in `.env` (and maybe not in `_ss_environment.php`) maybe it should still be in the output and not removed? It's probably ok if `_ss_environment.php` has higher priority (or whichever you want to do).